File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/analysis/QueryAnalysisUnchecked.scala
package dev.typr.foundationssc.docs.analysis
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object QueryAnalysisUnchecked:
  //start
  case class Stats(name: String, count: Int)

  // .unchecked() skips type checking entirely for this column
  val statsCodec: RowCodec[Stats] = RowCodec.builder[Stats]()
    .field(PgTypes.text)(_.name)
    .field(PgTypes.int4.unchecked())(_.count)
    .build(Stats.apply)
  //stop

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/analysis/QueryAnalysisUnchecked.kt
package dev.typr.foundationskt.docs.analysis

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class QueryAnalysisUnchecked {
    //start
    data class Stats(val name: String, val count: Int)

    // .unchecked() skips type checking entirely for this column
    val statsCodec: RowCodec<Stats> =
        RowCodec.builder<Stats>()
            .field(PgTypes.text, Stats::name)
            .field(PgTypes.int4.unchecked(), Stats::count)
            .build(::Stats)
    //stop
}
